handlers/auth.go

```diff
@@ -24,3 +24,19 @@ func Auth(secret string) gin.HandlerFunc {
 		c.Set("user", user)
 	}
 }
+
+func getUserFromJwt(webtoken, secret string) (*models.User, error) {
+	token, err := jwt.Parse(webtoken, func(token *jwt.Token)(interface{}, error){
```

Comment:
Ah, you're right, I missed the `jwt.ParseFromRequest/jwt.Parse` difference. The middle parts of these functions might be candidates for shared error-handling/user extraction, but I think this is all right for now.